CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -247,20 +250,45 @@ public static function updatePeriod($params) {
    * with other active periods within the same membership.
    *
    * @param $periodParams
+   *
    * @return bool
+   * @throws \CRM_Core_Exception
    */
   private static function doesOverlapWithOtherActivePeriods($periodParams) {
-    $membershipPeriod = self::getMembershipPeriodById($periodParams['id']);
+    if (!$periodParams['is_active']) {
+      return FALSE;
+    }
+
+    $periodID = CRM_Utils_Array::value('id', $periodParams, 0);
+    $membershipID = CRM_Utils_Array::value('membership_id', $periodParams, 0);
+    $periodNewStartDate = CRM_Utils_Array::value('start_date', $periodParams, '');
+    $periodNewEndDate = CRM_Utils_Array::value('end_date', $periodParams, '');
 
-    $periodNewStartDate = $periodParams['start_date'];
-    $periodNewEndDate = $periodParams['end_date'];
+    if (!empty($periodID)) {
+      $membershipPeriod = self::getMembershipPeriodById($periodID);
+      $membershipID = $membershipID ?: $membershipPeriod->membership_id;
+      $periodNewStartDate = $periodNewStartDate ?: $membershipPeriod->start_date;
+      $periodNewEndDate = $periodNewEndDate ?: $membershipPeriod->end_date;
+    }
+
+    if (empty($membershipID)) {
```

Comment:
as long as you fetch the period details as I suggested in my other comment, I think it would be pointless to check for empty membership_id and empty start date cases since they should always be there.